Fixed errors always empty

circulon opened the pull request:
Not really sure how this has stayed unnoticed for so long.

This fixes the "errors" put into session flash being removed at the beginning of the session and put into a view variable called "bag" for some reason.

The upshot of this is that the examples in the Validation documentation never work and the `ShareErrorsInSessionMiddleware` is always empty as it is loaded AFTER the session is established. 

Not really sure how to add a test for this.

josephmancuso committed:
Merge pull request #794 from eaguad1337/4.0

replace twitter with X

circulon committed:
Fixed errors always empty

circulon committed:
Fixed false pass if regex meta chars used

providing a list of regex meta charscters could cause a false pass for validation

riteshjha committed:
cookie settings from config

riteshjha committed:
fixed linting issue

josephmancuso committed:
added object check

josephmancuso committed:
Merge pull request #803 from MasoniteFramework/hotfix/database-queue-callable

added object check

eaguad1337 committed:
Merge pull request #718 from girardinsamuel/feat/663

[5.X] Add `logging` feature

eaguad1337 committed:
Revert "[5.X] Add `logging` feature"

eaguad1337 committed:
Merge pull request #807 from MasoniteFramework/revert-718-feat/663

Revert "[5.X] Add `logging` feature"

josephmancuso committed:
Merge pull request #799 from circulon/fix/strong_validator_special_chars

Fixed false pass if regex meta chars used

eaguad1337 committed:
fix: requirements.txt

circulon committed:
Fixe Redis Cache driver Issues

Refactored Redis Cache Driver
- Fixed internal cache not always loaded on first cace access
- added ability to define “timeout” for for item expiry
- - defaults to 1 month.  Previous hardcoded value was 10 years
- correctly store and unpack int types
- fix internal cache not removed if store was flushed
- fixed an issue where am imternal cache key would not be updated if it existed, but would be updated in the Redis store

josephmancuso committed:
Merge pull request #810 from MasoniteFramework/fix/requirements

fix: requirements.txt

circulon committed:
Merge branch '4.0' of https://github.com/MasoniteFramework/masonite into fix/redis_cache_issues

circulon committed:
Added key expiry to internal cache

Redis has no way to get key ttl enmasse to we oft for an on demand check and update the internal cache as required

circulon committed:
Updated tests for Redis Cache driver

circulon committed:
Fixed linting

circulon committed:
Cleaned up key expiry check

circulon commented:
@josephmancuso  @eaguad1337 
has there been any evaluation of this fix?

This issue may be affecting many devs without them even realising it

circulon committed:
Added opt-out for cookie encryption

circulon committed:
Fixed test not in class

josephmancuso committed:
Merge pull request #811 from circulon/fix/redis_cache_issues

Fix/redis cache issues

josephmancuso committed:
Merge pull request #814 from circulon/feature/cookie_encrypt_opt_out

opt-out of encrypt for a cookie

josephmancuso commented:
i dont think we can do this because the documentation says to use the bag helper .. and if we modify this to be done right i think it'll be a breaking change

josephmancuso committed:
Merge branch '4.0' into feature/800

josephmancuso committed:
Merge pull request #801 from riteshjha/feature/800

cookie settings from config

circulon commented:
Hi @josephmancuso yeah you are right this would revert to the old behaviour without a MessageBag.
That said the current behaviour is still broken.

I have worked out the issue and am in the process of addressing it.
I will alsp add tests for the Session middleware.

here is what I have so far:
> https://docs.masoniteproject.com/features/validation#displaying-errors-in-views
>Shows to use the  `session().get('errors')` method which always returns `None` 
>The `errors` variable in the View is only referenced as part of the `SharedErrorsInSession`

Additionally:
 It would be nice to have the `errors()` helper to be backwards compatible and return a MessageBag instance

Current errors in view behaviour:
```python
# file: index.html
...
session errors: {{ session().get('errors') }} {# always returns None #}                  
<br>                
session bag: {{ session().get('bag') }} {# always returns None #}
{# just proving there is no session var called 'bag' #}                  
<br>                  
view helper errors() : {# {{ errors().any() }} #} {# throws "errors() is undefined (when uncommented) #}                 
<br>                  
view helper bag() : {{ bag().errors() }} {# contains messages expected in errors() var #}
 ...
```

Expected behaviour:
```python
# file: index.html
...
session errors: {{ session().get('errors') }} {# expected to return None, an empty dict or even a MessageBag instance #}                  
<br>                
session bag: {{ session().get('bag') }} {# always returns None #}
{# just proving there is no session var called 'bag' #}                  
<br>                  
view helper errors() : {{ errors().any() }}  {# Expected value: True #}                 
<br>                  
view helper bag() : {# {{ bag().errors() }}  #} {# Throws 'bag()' is undefined  (qhwn uncommented) #}
 ...
```

I hope this clarifies things

circulon committed:
fixed session flash errors missing

session flash now contains errors key as it did previously.
The errors are now a MessageBag which aligns the Session with the View helper ‘errors()’

circulon committed:
Merge branch '4.0' of https://github.com/circulon/masonite into fix/errors_always_empty
